Tasks/KubeloginInstallerV0/utils.ts

```diff
@@ -0,0 +1,203 @@
+
+
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+
+import * as engine from 'artifact-engine/Engine';
+import * as providers from 'artifact-engine/Providers';
+import * as taskLib from 'azure-pipelines-task-lib/task';
+import * as toolLib from "azure-pipelines-tool-lib/tool";
+
+import octo = require('@octokit/rest')
+
+var process = require('process');
+var DecompressZip = require('decompress-zip');
+
+export const KUBELOGIN_REPO_OWNER = 'Azure';
+export const KUBELOGIN_REPO = 'kubelogin';
+
+const AGENT_TEMP_DIR = 'agent.tempDirectory';
+
+const octokit = new octo.Octokit();
+
+export type Platform =
+  | 'darwin-amd64'
+  | 'darwin-arm64'
+  | 'linux-amd64'
+  | 'linux-arm64'
+  | 'win-amd64';
+  
+export interface KubeloginRelease {
+  readonly version: string;
+  readonly platform: Platform;
+  readonly name: string;
+  readonly releaseUrl: string;
+  readonly checksumUrl: string;
+}
+
+export function resolvePlatform(): Platform {
+  const platform = process.platform;
+  const arch = process.arch;
+  if (platform === 'darwin' && arch === 'x64') {
+    return 'darwin-amd64';
+  } else if (platform === 'darwin' && arch === 'arm64') {
+    return 'darwin-arm64';
+  } else if (platform === 'linux' && arch === 'x64') {
+    return 'linux-amd64';
+  } else if (platform === 'linux' && arch === 'arm64') {
+    return 'linux-arm64';
+  } else if (platform === 'win32' && arch === 'x64') {
+    return 'win-amd64';
+  } else {
+    throw new Error(`Unsupported platform: ${platform}-${arch}`);
+  }
+}
+
+export function isLatestVersion(version: string): boolean {
+    const v = version.toLowerCase();
+    return v === 'latest' || v === '*' || v === '';
+}
+  
+export async function getLatestVersionTag(): Promise<string> {
+  const tag_name = await octokit.repos.getLatestRelease({
```

Comment:
Does the octokit support proxies? Is it compatible with the way we set up proxies?

Reply:
I replaced octokit with WebClient from azure-arm-rest-v2 library. It initialize required configuration for a proxy server